### PR TITLE
Add changelog for node release 8.6

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by teodanciu** [comment]
@lehins  Not sure if all these changes are worth their spot in the changelog, what do you think?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,19 @@ the `CHANGELOG.md` for each individual package for any changes relevant for deve
 If you are looking for the Ledger Releasing and Versioning Process then you can find it in
 [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd).
 
+## 8.6
+
+- Performance tweaks related to `Umap.size`
```
Comment (lehins) [inline CHANGELOG.md]: This is not a user facing change, also it had no measurable impact on performance
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,19 @@ the `CHANGELOG.md` for each individual package for any changes relevant for deve
...
+- Prevent updating protocol version with PParamUpdate
+- Improve native script handling:
```
Comment (lehins) [inline CHANGELOG.md]:
```suggestion
- Slight performance improvements to native script handling:
```

**Review by lehins — APPROVED** [review]
Thank you for doing this!

**Pull request merged**